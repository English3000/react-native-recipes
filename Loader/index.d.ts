/** To resolve TypeScript errors */
declare module "*.png"