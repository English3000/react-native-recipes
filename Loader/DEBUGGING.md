1. If `react-native run-android` has a build error...
  * Don't debug. Just start anew.
    > i.e. If it doesn't have to do w/ my code, it's not my problem.

2. If simulator complains about `@babel/regenerator-runtime`...
  * Remove `node_modules` & any `lock` files. `npm i`
    > #setup

3. If simulator has a `500` error...
  * Close it. Kill the terminal server. Re-`run`
    > happened when I updated `node_modules`