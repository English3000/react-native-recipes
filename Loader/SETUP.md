### Notes
* On simulators, need to _enable_ hot reloading.
* For CSS color names, https://www.rapidtables.com/web/color/

### Commands
```bash
# To create new project w/ TypeScript
react-native init <> --template=typescript
# To update Jest snapshots
npm test -- -u
```

### Testing
Jest snapshots serve as a baseline (TDD).