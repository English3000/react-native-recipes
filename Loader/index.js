// @ts-check
import React from "react"
import { AppRegistry, View, Text } from "react-native"
import { name } from "./app.json"

export default function Loader(){
  return (
    <View>
      <Text>working!</Text>
    </View>
  )
}

AppRegistry.registerComponent(name, () => Loader)