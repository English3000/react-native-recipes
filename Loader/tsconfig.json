{
  "compilerOptions": {
    "jsx":              "react",
    "lib":              ["es6"],
    "moduleResolution": "node",
    "target":           "esnext",

    "allowJs":                      true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop":              true,
    "isolatedModules":              true,
    "noEmit":                       true,
    "resolveJsonModule":            true,
    "strict":                       true,
  },
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js"]
}
