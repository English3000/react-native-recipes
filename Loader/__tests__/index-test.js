import React from "react"
import "react-native"
import { create } from "react-test-renderer"
import Loader from "../index"

it("renders", () => {
  expect(
    create(<Loader />).toJSON()
  ).toMatchSnapshot()
})