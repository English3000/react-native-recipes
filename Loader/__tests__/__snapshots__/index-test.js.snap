// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View>
  <Text>
    working!
  </Text>
</View>
`;
